sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/CatalogFileIndex.scala

```diff
@@ -94,6 +93,19 @@ class CatalogFileIndex(
     }
   }
 
+  private def listPartitionsByFilter(
```

Comment:
this code is duplicated from the one in `PruneHiveTablePartitions`, and hopefully we can remove it once we combine `PruneHiveTablePartitions` and `PruneFileSourcePartitions`.

Reply:
Shall we use the same name, `private def prunePartitions`, explicitly?
When we consolidate both code later, that will reduce the difference.